airflow/task/task_runner/standard_task_runner.py

```diff
@@ -110,7 +110,6 @@ def terminate(self):
             return
 
         # Reap the child process - it may already be finished
-        _ = self.return_code(timeout=0)
```

Comment:
If we _don't_ do this (i.e. if `self.return_code` is never called) then we end up with a Zombie process (not to be confused with a zombie Task Instance) https://www.howtogeek.com/119815/htg-explains-what-is-a-zombie-process-on-linux/

I don't see that this causes a problem? `return_code()` is already implemented in such a way that if the process has already exited it doesn't do anything.

Is there a problem you are trying to solve here?


Looking at what `return_code()` does, L102 might actually be the problem, and we should only clear the process if we actually got a return code back (i.e. if the process has exited).

Reply:
No problem. Just fixing the test but we can as well go with the solution I have at https://github.com/apache/airflow/pull/17581.

Reply:
Thanks for the link!